src/components/IFrame.js

```diff
@@ -97,38 +190,102 @@ const propTypes = {
     }).isRequired,
 };
 
-function OldDotIFrame({session}) {
-    const [oldDotURL, setOldDotURL] = useState('https://staging.expensify.com');
+// Gets the routeName and params of the top most central pane route if it is an iframe route.
+// If the top most central pane route is not an iframe route, then it returns undefined.
+function getIframeRouteNameAndParams(state) {
+    const routeName = getTopMostCentralPaneRouteName(state);
+    if (routeName !== undefined && routeName !== 'Report') {
+        const topMostCentralPane = lodashFindLast(state.routes, (route) => route.name === 'CentralPaneNavigator');
+        if (topMostCentralPane && topMostCentralPane.state) {
+            const params = topMostCentralPane.state.routes[topMostCentralPane.state.index || 0].params;
+            return {routeName, params, index: state.index};
+        }
 
-    useEffect(() => {
-        setOldDotURL(`https://expensify.com.dev/${getOldDotURL(window.location.href)}`);
+        const params = topMostCentralPane.params;
+        if (params) {
+            return {routeName, params: params.params, index: state.index};
+        }
+    }
+    return {routeName: undefined, params: undefined, index: state.index};
+}
+
+// TODO: use proper URL
+const BASE_IFRAME_URL = 'http://localhost:3000';
```

Comment:
I assume we need to change this before we merge?

Reply:
I think we have a constant for this that changes depending on the environment